src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -381,3 +527,53 @@ def __await__(self):
 
     async def wait_for_connection(self) -> None:
         pass
+
+
+class UnaryStreamCallResponseIterator(_base_call.UnaryStreamCall):
+    """UnaryStreamCall class wich uses an alternative response iterator."""
+    _call: _base_call.UnaryStreamCall
+    _response_iterator: AsyncIterable[ResponseType]
+
+    def __init__(self, call: _base_call.UnaryStreamCall,
+                 response_iterator: AsyncIterable[ResponseType]) -> None:
+        self._response_iterator = response_iterator
+        self._call = call
+
+    def cancel(self) -> bool:
+        return self._call.cancel()
+
+    def cancelled(self) -> bool:
+        return self._call.cancelled()
+
+    def done(self) -> bool:
+        return self._call.done()
+
+    def add_done_callback(self, callback) -> None:
+        self._call.add_done_callback(callback)
+
+    def time_remaining(self) -> Optional[float]:
+        return self._call.time_remaining()
+
+    async def initial_metadata(self) -> Optional[MetadataType]:
+        return await self._call.initial_metadata()
+
+    async def trailing_metadata(self) -> Optional[MetadataType]:
+        return await self._call.trailing_metadata()
+
+    async def code(self) -> grpc.StatusCode:
+        return await self._call.code()
+
+    async def details(self) -> str:
+        return await self._call.details()
+
+    async def debug_error_string(self) -> Optional[str]:
+        return await self._call.debug_error_string()
+
+    def __aiter__(self):
+        return self._response_iterator.__aiter__()
+
+    async def wait_for_connection(self) -> None:
+        return await self._call.wait_for_connection()
+
+    async def read(self) -> ResponseType:
+        return await self._call.read()
```

Comment:
I would say that this function is never used since the `read` at InterceptedCall level uses [1] under the hood the iterator.

[1] https://github.com/grpc/grpc/pull/22678/files#diff-173e8bd5d2219ba109f86237680068a4R481